doc/sphinx/practical-tools/utilities.rst

```diff
@@ -559,43 +629,73 @@ To build, say, two targets foo.vo and bar.vo in parallel one can use
 
 .. note::
 
-  ``make foo.vo bar.vo -j`` has a different meaning for the make
+  ``make foo.vo bar.vo -j`` has a different meaning for the ``make``
   utility, in particular it may build a shared prerequisite twice.
 
-
-.. note::
-
-  For users of coq_makefile with version < 8.7
-
-  + Support for "subdirectory" is deprecated. To perform actions before
-    or after the build (like invoking ``make`` on a subdirectory) one can hook
-    in pre-all and post-all extension points.
-  + ``-extra-phony`` and ``-extra`` are deprecated. To provide additional target
-    (``.PHONY`` or not) please use ``CoqMakefile.local``.
-
 .. note::
 
   Due to limitations with the compilation chain, makefiles generated
   by ``coq_makefile`` won't correctly compile OCaml plugins with OCaml
   < 4.07.0 when using more than one job (``-j N`` for ``N > 1``).
 
+
+The grammar of _CoqProject
+++++++++++++++++++++++++++
+The parser of ``coq_makefile`` uses the following syntax:
+
+  .. prodn::
+     CoqProject ::= {* {| @blank | @comment | @quoted_string | @unquoted_string } }
+     blank ::= {| space | horizontal_tab | newline }
+     comment ::= # {* comment_char } newline
+     quoted_string ::= " {* quoted_char } "
+     unquoted_string ::= string_start_char {* unquoted_char }
+
+Where the following definitions apply:
+
+* :n:`space`, :n:`horizontal_tab` and :n:`newline` stand for the corresponding
+  ASCII characters.
+* :n:`comment_char` is the set of all characters except :n:`newline`.
+* :n:`quoted_char` is the set of all characters except ``"``.
+* :n:`string_start_char` is the set of all characters except those that match :n:`@blank`, or are ``"`` or ``#``.
+* :n:`unquoted_char` is the set of all characters except those that match :n:`@blank` or are ``#``.
+
+The parser produces a list of strings in the same order as they were
+encountered in ``_CoqProject``. All :n:`@blank` and :n:`@comment` tokens
+get ignored and the double quotes of :n:`@quoted_string` tokens are removed as
+well.
+
+The current approach has a few limitations: Double quotes are only special
+characters at the start of a string. And for lack of an escaping mechanism, it
+is currently impossible to encode the following kinds of strings in a
+``_CoqProject`` file:
+
+* strings starting with ``"``
+* strings starting with ``#`` and containing ``"``
+* strings containing both whitespace and ``"``
```

Comment:
I didn't want to decide how to parse the escape sequences, and when tokens should "stay together" or "split apart" on (non-)escaped whitespace. I think I'd first need a specification of how a new escaping/quoting mechanism should behave (as seen from the user's p.o.v.) before I can implement it.
Maybe discuss and resolve these limitations in a separate thread (issue/PR/cep/Zulip)? How about I sketch how the new parser might behave, then "we" (whoever is interested) can discuss that and in the end I implement it?

I think it would be best, if the new behaviour wouldn’t treat `-arg` as a special case, as it does now. (Regarding single-quotes)

About quoted/unquoted strings. I wanted to say something like: After splitting the _CoqProject file into tokens, it is impossible to have tokens of the following form: ...

Reply:
It's your choice whether to add an escape mechanism.  I'd consider how useful it is to users to support these before spending time on it--e.g. how often do users want a `"` at the beginning of a string (or escaped at all)?  You could create a short proposal as a new issue if you wish and perhaps include `@gares` in the description to get his attention.

Hope you will tweak this wording to be clearer about quoted/unquoted strings.